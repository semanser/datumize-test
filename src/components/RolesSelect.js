import React from 'react'
import Select from 'react-select'
import 'react-select/dist/react-select.css'

const RolesSelect = props => (
  <Select
    className="select"
    name="form-field-name"
    value={props.selectedProjectId}
    clearable={false}
    onChange={selected => {props.changeProject(selected.value)}}
    options={props.projects.map(project => ({
      label: project.name,
      value: project.id,
    }))}
  />
)

export default RolesSelect

