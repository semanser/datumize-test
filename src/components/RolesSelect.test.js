import React from 'react'
import ReactDOM from 'react-dom'
import ShallowRenderer from 'react-test-renderer/shallow'
import RolesSelect from './RolesSelect'

const renderer = new ShallowRenderer()

const projects = [
  {
    id: 1,
    name: 'test project 1',
  },
  {
    id: 2,
    name: 'test project 2',
  },
  {
    id: 3,
    name: 'test project 3',
  },
]

describe('RolesSelect Component', () => {
  it('renders component without crashing', () => {
    const div = document.createElement('div')

    const props = {
      projects,
    }

    const tree = renderer.render(<RolesSelect {...props}/>)
    expect(tree).toMatchSnapshot()
  })
})

