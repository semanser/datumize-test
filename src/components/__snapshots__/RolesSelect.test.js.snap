// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RolesSelect Component renders component without crashing 1`] = `
<Select
  arrowRenderer={[Function]}
  autosize={true}
  backspaceRemoves={true}
  backspaceToRemoveMessage="Press backspace to remove {label}"
  className="select"
  clearAllText="Clear all"
  clearRenderer={[Function]}
  clearValueText="Clear value"
  clearable={false}
  closeOnSelect={true}
  deleteRemoves={true}
  delimiter=","
  disabled={false}
  escapeClearsValue={true}
  filterOptions={[Function]}
  ignoreAccents={true}
  ignoreCase={true}
  inputProps={Object {}}
  isLoading={false}
  joinValues={false}
  labelKey="label"
  matchPos="any"
  matchProp="any"
  menuBuffer={0}
  menuRenderer={[Function]}
  multi={false}
  name="form-field-name"
  noResultsText="No results found"
  onBlurResetsInput={true}
  onChange={[Function]}
  onCloseResetsInput={true}
  onSelectResetsInput={true}
  openOnClick={true}
  optionComponent={[Function]}
  options={
    Array [
      Object {
        "label": "test project 1",
        "value": 1,
      },
      Object {
        "label": "test project 2",
        "value": 2,
      },
      Object {
        "label": "test project 3",
        "value": 3,
      },
    ]
  }
  pageSize={5}
  placeholder="Select..."
  removeSelected={true}
  required={false}
  rtl={false}
  scrollMenuIntoView={true}
  searchable={true}
  simpleValue={false}
  tabSelectsValue={true}
  trimFilter={true}
  value={undefined}
  valueComponent={[Function]}
  valueKey="value"
/>
`;
